### PR TITLE
feat(components): Main navigation and megadropdown

### PR DESCRIPTION
**Pull request by gfellerph**
Added prototypes for the main navigation and the megadropdown for structure testing. Styles, tests and documentation is still missing.

### CODE REVIEW

File: packages/components/src/components/post-megadropdown-toggle/post-megadropdown-toggle.tsx
Comment (alizedebray) [inline packages/components/src/components/post-megadropdown-toggle/post-megadropdown-toggle.tsx:1]: Can we rename this to `post-megadropdown-trigger` to align naming with the `post-collapsible-trigger`?

File: packages/components/src/components/post-megadropdown-toggle/post-megadropdown-toggle.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+@Component({
+  tag: 'post-megadropdown-toggle',
+  shadow: true,
+  styleUrl: './post-megadropdown-toggle.scss',
+})
+export class PostMegadropdownToggle {
+  @Event() postMegadropdownToggled: EventEmitter;
+
+  @State() toggled = false;
```
Comment (alizedebray) [inline packages/components/src/components/post-megadropdown-toggle/post-megadropdown-toggle.tsx:11]: Is this state necessary? It does not seem to be used anywhere.

File: packages/components/src/components/post-megadropdown-toggle/post-megadropdown-toggle.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+  private handleClick() {
+    this.toggled = !this.toggled;
+    this.postMegadropdownToggled.emit();
+  }
+
+  render() {
+    return (
+      <Host slot="post-megadropdown-toggle">
+        <button onClick={() => this.handleClick()}>
```
Comment (alizedebray) [inline packages/components/src/components/post-megadropdown-toggle/post-megadropdown-toggle.tsx:21]: This button should probably have ARIA attributes.

File: packages/components/src/components/post-megadropdown-toggle/post-megadropdown-toggle.scss
Comment (alizedebray) [inline packages/components/src/components/post-megadropdown-toggle/post-megadropdown-toggle.scss:1]: This probably need an: 
```scss
:host {
  display: inline-block;
}
```

File: packages/components/src/components/post-mainnavigation/post-mainnavigation.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+@Component({
+  tag: 'post-mainnavigation',
+  shadow: false,
+  styleUrl: './post-mainnavigation.scss',
+})
+export class PostMainnavigation {
+  /**
+   * Gets emitted when a user closes the main navigation on mobile
+   */
+  @Event() postMainNavigationClosed: EventEmitter;
```
Comment (alizedebray) [inline packages/components/src/components/post-mainnavigation/post-mainnavigation.tsx:12]: To align with other events: 
```suggestion
  @Event() postToggle: EventEmitter;
```

File: packages/components/src/components/post-megadropdown-toggle/post-megadropdown-toggle.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import { Component, Event, EventEmitter, Host, State, h } from '@stencil/core';
+
+@Component({
+  tag: 'post-megadropdown-toggle',
+  shadow: true,
+  styleUrl: './post-megadropdown-toggle.scss',
+})
+export class PostMegadropdownToggle {
+  @Event() postMegadropdownToggled: EventEmitter;
```
Comment (alizedebray) [inline packages/components/src/components/post-megadropdown-toggle/post-megadropdown-toggle.tsx:9]: Same here:
```suggestion
  @Event() postToggle: EventEmitter;
```

**Review by alizedebray — CHANGES_REQUESTED** [review]

File: packages/components/src/index.html
[the remark this answers is not in the document]
Comment (gfellerph) [reply packages/components/src/index.html:1]: Can we keep the changes here? The markup structure is not documented anywhere else for the moment. We can remove it as soon as the header is properly documented in storybook.

File: packages/components/src/components/post-megadropdown-toggle/post-megadropdown-toggle.tsx (hunk 1 above)
Comment (gfellerph) [reply packages/components/src/components/post-megadropdown-toggle/post-megadropdown-toggle.tsx:11]: Added usage for `aria-expanded`.

File: packages/components/src/index.html
[the remark this answers is not in the document]
Comment (alizedebray) [reply packages/components/src/index.html:1]: Ok!

**Review by alizedebray — APPROVED** [review]

**Pull request merged**